packages/apps/shopify-app-remix/src/server/authenticate/admin/scope/types.ts

```diff
@@ -1,14 +1,166 @@
+/**
+ * The Scopes API enables embedded apps and extensions to request merchant consent for access scopes.
+ */
 export interface ScopesApiContext {
+  /**
+   * Queries Shopify for the scopes for this app on this shop
+   *
+   * @returns {ScopesDetail} The scope details.
+   *
+   * @example
+   * <caption>Query for scopes details.</caption>
+   * <description>Call `scopes.query` to get scope details.</description>
+   * ```ts
+   * // /app/routes/app.query.tsx
+   *import type { LoaderFunctionArgs } from "@remix-run/node";
+   *import { json } from "@remix-run/node";
+   *
+   *import { authenticate } from "../shopify.server";
+   *
+   *export async function loader({
+   *  request,
+   *}: LoaderFunctionArgs) {
+   *  const { scopes } = await authenticate.admin(request);
+   *
+   *  const response = await scopes.query();
+   *  return json({scopesDetail : response});
+   *}
+   * ```
+   * ```ts
+   * // /app/shopify.server.ts
+   * import { LATEST_API_VERSION, shopifyApp } from "@shopify/shopify-app-remix/server";
+   *
+   * const shopify = shopifyApp({
+   *   // ...etc
+   * });
+   * export default shopify;
+   * export const authenticate = shopify.authenticate;
+   * ```
+   */
   query: () => Promise<ScopesDetail>;
-  request: (scopes: string[]) => Promise<void>;
-  revoke: (scopes: string[]) => Promise<RevokeResponse>;
-}
 
-export interface RevokeResponse {
-  revoked: string[];
+  /**
+   * Requests the merchant to grant the provided scopes for this app on this shop
+   *
+   * **Warning:** This method performs a server-side redirect.
+   *
+   * @example
+   * <caption>Request consent from the merchant to grant the provided scopes for this app.</caption>
+   * <description>Call `scopes.request` to request optional scopes.</description>
+   * ```ts
+   * // /app/routes/app.request.tsx
+   *import type { LoaderFunctionArgs } from "@remix-run/node";
+   *import { json } from "@remix-run/node";
+   *
+   *import { authenticate } from "../shopify.server";
+   *import { AuthScopes } from "@shopify/shopify-api";
+   *export async function loader({
```

Comment:
From the context below, it looks like this is being set up more specifically as an `action`: `request.formData()`, etc.

I think the `query` makes sense in a `loader`, but `request` perhaps best makes sense as part of a resource route's `action`, and perhaps the same for `revoke`. Worth testing for those